File: kyonggiyo-core/src/main/java/kyonggiyo/domain/candidate/Candidate.java
package kyonggiyo.domain.candidate;

import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import kyonggiyo.domain.BaseEntity;
import kyonggiyo.domain.restaurant.Address;
import kyonggiyo.domain.restaurant.Restaurant;
import kyonggiyo.domain.restaurant.RestaurantCategory;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "candidates")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Candidate extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Enumerated(EnumType.STRING)
    private RestaurantCategory category;

    private String contact;

    @Embedded
    private Address address;

    private String reason;

    @Enumerated(EnumType.STRING)
    private Status status = Status.WAITING;

    private Long requesterId;

    @Builder
    private Candidate(String name,
                      RestaurantCategory category,
                      String contact,
                      String address,
                      double lat,
                      double lng,
                      String reason,
                      Long requesterId) {
        this.name = name;
        this.category = category;
        this.contact = contact;
        this.address = new Address(address, lat, lng);
        this.reason = reason;
        this.requesterId = requesterId;
    }

    public void accept() {
        this.status = Status.ACCEPTED;
    }

    public void updateName(String name) {
        this.name = name;
    }

    public void updateCategory(RestaurantCategory category) {
        this.category = category;
    }

    public void updateContact(String contact) {
        this.contact = contact;
    }

    public void updateAddress(String address, double lat, double lng) {
        this.address = new Address(address, lat, lng);
    }

    public void updateReason(String reason) {
        this.reason = reason;
    }

    public Restaurant toRestaurant() {
        return Restaurant.builder()
                .name(name)
                .category(category)
                .contact(contact)
                .address(address.getAddress())
                .lat(address.getLat())
                .lng(address.getLng())
                .reason(reason)
                .build();
    }

}

File: kyonggiyo-core/src/main/java/kyonggiyo/application/candidate/domain/entity/Candidate.kt
package kyonggiyo.application.candidate.domain.entity

import jakarta.persistence.*
import kyonggiyo.application.candidate.domain.vo.Status
import kyonggiyo.domain.BaseEntity
import kyonggiyo.domain.restaurant.Address
import kyonggiyo.domain.restaurant.Restaurant
import kyonggiyo.domain.restaurant.RestaurantCategory

@Entity
@Table(name = "candidates")
class Candidate(
    name: String,
    category: RestaurantCategory,
    contact: String,
    reason: String,
    requesterId: Long,
    address: String,
    lat: Double,
    lng: Double
): BaseEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    @Column(nullable = false)
    val requesterId: Long = requesterId

    @Column(nullable = false)
    var name: String = name
        protected set

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    var category: RestaurantCategory = category
        protected set

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    var status = Status.WAITING
        protected set

    @Column(nullable = false)
    @Embedded
    var address = Address(address, lat, lng)
        protected set

    @Column(nullable = false)
    var contact: String = contact
        protected set

    @Column(nullable = false)
    var reason: String = reason
        protected set

    fun accept() {
        this.status = Status.ACCEPTED
    }

    fun updateName(name: String) {
        this.name = name
    }

    fun updateCategory(category: RestaurantCategory) {
        this.category = category
    }

    fun updateContact(contact: String) {
        this.contact = contact
    }

    fun updateAddress(address: String, lat: Double, lng: Double) {
        this.address = Address(address, lat, lng)
    }

    fun updateReason(reason: String) {
        this.reason = reason
    }

    fun toRestaurant(): Restaurant {
        return Restaurant(
            name = name,
            category = category,
            contact = contact,
            address = address,
            lat = address.lat,
            lng = address.lng,
            reason = reason
        )
    }
}
